spec/defines/service_spec.rb

```diff
@@ -0,0 +1,75 @@
+require 'spec_helper'
+describe 'pam::service', :type => :define do
+  context 'on a RedHat OS' do
+    let(:facts) {
+      {
+        :osfamily                   => 'RedHat',
+        :operatingsystemmajrelease  => '5',
+      }
+    }
+    context 'with no parameters' do
+      let(:title) { 'test' }
+      it { should contain_class('pam') }
+      it { should contain_file('pam.d-service-test').with(
+        'ensure'  => 'file',
+        'owner'   => 'root',
+        'group'   => 'root',
+        'mode'    => '0644',
+        'content' => nil
+      ) }
+    end
+    context 'when absent' do
+      let(:title) { 'test' }
+      let :params do
+          {
+            :ensure  => 'absent'
+          }
+        end
+      it { should contain_file('pam.d-service-test').with(
+        'ensure'  => 'absent'
+      ) }
+    end
+    context 'when given content' do
+      let(:title) { 'test' }
+      let :params do
+          {
+            :content  => 'session required pam_permit.so'
+          }
+        end
+      it { should contain_file('pam.d-service-test').with_content(
+        'session required pam_permit.so'
+      ) }
+    end
+    context 'when given an array of lines' do
+      let(:title) { 'test' }
+      let :params do
+          {
+            :lines  => [
+              '@include common-auth',
+              '@include common-account',
+              'session required pam_permit.so',
+              'session required pam_limits.so'
+            ]
+          }
+        end
+      it { should contain_file('pam.d-service-test').with_content(
```

Comment:
Since the ordering of the lines is important to PAM we should check that the entire file's contents match instead of per line checks. Here's an example https://github.com/ghoneycutt/puppet-module-dnsclient/blob/master/spec/classes/init_spec.rb#L18-26